apps/webservice/src/app/[workspaceSlug]/(appv2)/resources/(raw)/[resourceId]/deployments/ResourceDeploymentTable.tsx

```diff
@@ -0,0 +1,172 @@
+import type { StatsOrder } from "@ctrlplane/validators/deployments";
+import type { JobCondition } from "@ctrlplane/validators/jobs";
+import { useMemo } from "react";
+import { useSearchParams } from "next/navigation";
+import { subDays } from "date-fns";
+import _ from "lodash";
+
+import { Card } from "@ctrlplane/ui/card";
+import {
+  Table,
+  TableBody,
+  TableHead,
+  TableHeader,
+  TableRow,
+} from "@ctrlplane/ui/table";
+import {
+  ColumnOperator,
+  ComparisonOperator,
+  DateOperator,
+  FilterType,
+} from "@ctrlplane/validators/conditions";
+import { StatsColumn } from "@ctrlplane/validators/deployments";
+import { JobFilterType } from "@ctrlplane/validators/jobs";
+
+import { TableSortHeader } from "~/app/[workspaceSlug]/(appv2)/_components/TableSortHeader";
+import { api } from "~/trpc/react";
+import { ResourceDeploymentRow } from "./ResourceDeploymentRow";
+import { ResourceDeploymentRowSkeleton } from "./ResourceDeploymentRowSkeleton";
+
+type ResourceDeploymentsTableProps = { resource: SCHEMA.Resource };
+
+const getFilter = (
+  resourceId: string,
+  startDate: Date,
+  endDate: Date,
+): JobCondition => {
+  const resourceFilter: JobCondition = {
+    type: JobFilterType.JobResource,
+    operator: ColumnOperator.Equals,
+    value: resourceId,
+  };
+
+  const startDateFilter: JobCondition = {
+    type: FilterType.CreatedAt,
+    operator: DateOperator.AfterOrOn,
+    value: startDate.toISOString(),
+  };
+
+  const endDateFilter: JobCondition = {
+    type: FilterType.CreatedAt,
+    operator: DateOperator.BeforeOrOn,
+    value: endDate.toISOString(),
+  };
+
+  return {
+    type: FilterType.Comparison,
+    operator: ComparisonOperator.And,
+    conditions: [resourceFilter, startDateFilter, endDateFilter],
+  };
+};
+
+export const ResourceDeploymentsTable: React.FC<
+  ResourceDeploymentsTableProps
+> = ({ resource }) => {
+  const { workspaceId } = resource;
+
+  const params = useSearchParams();
+  const orderByParam = params.get("order-by");
+  const orderParam = params.get("order");
+
+  const endDate = useMemo(() => new Date(), []);
+  const startDate = subDays(endDate, 14);
+  const timezone = Intl.DateTimeFormat().resolvedOptions().timeZone;
+
+  const filter = getFilter(resource.id, startDate, endDate);
+
+  const triggersQ = api.job.config.byWorkspaceId.list.useQuery({
+    workspaceId,
+    filter,
+    limit: 100,
+  });
+
+  const triggers = triggersQ.data ?? [];
+
+  const latestTriggersByDeployment = _.chain(triggers)
+    .groupBy((t) => t.release.deploymentId)
+    .map((groupedTriggers) => _.maxBy(groupedTriggers, (t) => t.job.startedAt))
+    .filter((t) => t != null)
+    .value();
+
+  const statsQ = api.deployment.stats.byWorkspaceId.useQuery({
+    workspaceId,
+    resourceId: resource.id,
+    startDate,
+    endDate,
+    timezone,
+    orderBy: orderByParam != null ? (orderByParam as StatsColumn) : undefined,
+    order: orderParam != null ? (orderParam as StatsOrder) : undefined,
```

Comment:
```suggestion
    orderBy: orderByParam ?? undefined,
    order: orderParam ?? undefined,
```

Ideally otherwise

```
    orderBy:(orderByParam as StatsColumn | null) ?? undefined,
    order: (orderParam as StatsOrder | null) ?? undefined,
 ```